### PR TITLE
Automated cherry pick of #915: Support configuring sysctls for Pods and enable net.*

### PR DESCRIPTION
**Pull request by aylei**
Cherry pick of #915 on release-1.0.

#915: Support configuring sysctls for Pods and enable net.*

### CODE REVIEW

File: pkg/apis/pingcap.com/v1alpha1/types.go [hunk 1]
```diff
@@ -90,9 +90,10 @@ type TidbClusterSpec struct {
 	TiKV            TiKVSpec            `json:"tikv,omitempty"`
 	TiKVPromGateway TiKVPromGatewaySpec `json:"tikvPromGateway,omitempty"`
 	// Services list non-headless services type used in TidbCluster
-	Services        []Service                            `json:"services,omitempty"`
-	PVReclaimPolicy corev1.PersistentVolumeReclaimPolicy `json:"pvReclaimPolicy,omitempty"`
-	Timezone        string                               `json:"timezone,omitempty"`
+	Services         []Service                            `json:"services,omitempty"`
+	PVReclaimPolicy  corev1.PersistentVolumeReclaimPolicy `json:"pvReclaimPolicy,omitempty"`
+	Timezone         string                               `json:"timezone,omitempty"`
+	MaxFailoverCount int32                                `json:"maxFailoverCount,omitempty"`
```
Comment (tennix) [inline pkg/apis/pingcap.com/v1alpha1/types.go]: Why add `MaxFailoverCount` in `TidbCluster`?

Comment (aylei) [reply pkg/apis/pingcap.com/v1alpha1/types.go]: Sorry, that's a typo when I fix the merge conflict in TiDBSpec

Comment (aylei) [reply pkg/apis/pingcap.com/v1alpha1/types.go]: Looks like this field is actually exists in master but is not used anywhere...

https://github.com/aylei/tidb-operator/blob/automated-cherry-pick-of-%23915-upstream-release-1.0/pkg/apis/pingcap.com/v1alpha1/types.go#L96

**Pull request merged**